doc/source/whatsnew/v0.25.0.rst

```diff
@@ -33,6 +33,9 @@ Backwards incompatible API changes
 
 .. _whatsnew_0250.api_breaking.utc_offset_indexing:
 
+- Comparing :class:`~Timestamp.Timestamp` with unsupported objects now returns ``NotImplemented`` instead of raising ``TypeError``. This implies that unsupported rich comparisons are delegated to the other object, and is now consistent with Python 3 behavior for ``datetime`` objects (:issue:`24011`)
```

Comment:
This should just be :class:`\Timestamp\` I think.

Otherwise, I think the description is sufficient. You *might* link to the python docs on NotImplemented, in case people are confused and think that we return that value to the end user?

Reply:
I could not build the documentation, so I don't have ways to check whether Timestamp or Timestamp.Timestamp is required. I will try to fix this asap. Meanwhile, I made the change, hoping ":py:obj:`NotImplemented`" is correct ;-)

Reply:
I think it's :class:`\Timestamp\` and :class:\`NotImplemented\`.

Reply:
I think `NotImplemented` is an object, not a class (see https://github.com/python/cpython/blob/4d61e6e3b802399be62a521d6fa785698cb670b5/Objects/object.c#L1730). Am I wrong?

Reply:
or maybe constant/special value?

https://docs.python.org/3/library/constants.html